document-store/src/integrationTest/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutorTest.java

```diff
@@ -0,0 +1,306 @@
+                            EQ,
+                            ConstantExpression.of(700007)))
+                    .build())
+            .addSort(IdentifierExpression.of("props.brand"), ASC)
+            .addSort(IdentifierExpression.of("item"), ASC)
+            .addSort(IdentifierExpression.of("props.seller.address.city"), ASC)
+            .build();
+
+    Iterator<Document> resultDocs = collection.find(query);
+    assertDocsEqual(resultDocs, "mongo/filter_on_nested_fields_response.json");
+  }
+
+  @Test
+  public void testAggregateEmpty() throws IOException {
+    Query query = Query.builder().build();
+
+    Iterator<Document> resultDocs = collection.aggregate(query);
+    assertSizeEqual(resultDocs, "mongo/collection_data.json");
+  }
+
+  @Test
+  public void testAggregateSimple() throws IOException {
+    Query query =
+        Query.builder()
+            .addSelection(AggregateExpression.of(SUM, ConstantExpression.of(1)), "count")
+            .build();
+
+    Iterator<Document> resultDocs = collection.aggregate(query);
+    assertDocsEqual(resultDocs, "mongo/count_response.json");
+  }
+
+  @Test
+  public void testAggregateWithFiltersAndOrdering() throws IOException {
+    Query query =
+        Query.builder()
+            .addSelection(
+                AggregateExpression.of(
+                    SUM,
+                    FunctionExpression.builder()
+                        .operand(IdentifierExpression.of("price"))
+                        .operator(MULTIPLY)
+                        .operand(IdentifierExpression.of("quantity"))
+                        .build()),
+                "total")
+            .addSelection(AggregateExpression.of(FIRST, IdentifierExpression.of("item")), "item")
+            .addAggregation(IdentifierExpression.of("item"))
+            .addSort(IdentifierExpression.of("total"), DESC)
+            .setAggregationFilter(
+                LogicalExpression.builder()
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("total"), GTE, ConstantExpression.of(25)))
+                    .operator(AND)
+                    .operand(
+                        RelationalExpression.of(
+                            IdentifierExpression.of("total"), LTE, ConstantExpression.of(99)))
+                    .build())
+            .setFilter(
+                RelationalExpression.of(
+                    IdentifierExpression.of("quantity"), NEQ, ConstantExpression.of(10)))
+            .setLimit(10)
+            .setOffset(0)
+            .build();
+
+    Iterator<Document> resultDocs = collection.aggregate(query);
+    assertDocsEqual(resultDocs, "mongo/sum_response.json");
+  }
+
+  @Test
+  public void testAggregateWithNestedFields() throws IOException {
+    Query query =
+        Query.builder()
+            .addSelection(
+                AggregateExpression.of(FIRST, IdentifierExpression.of("item")), "first_item")
```

Comment:
How easy would this be to do in `Postgres`? It is group by, right?

Reply:
Yup. Just found that this is feasible in Postgres (Ref.: https://stackoverflow.com/a/7630564)